expr/plan.go

```diff
@@ -238,124 +335,49 @@ func newplanFunc(e *expr, fn Func, from, to uint32, stable bool, reqs []Req) ([]
 	// for any provided keyword args, verify that they are what the function stipulated
 	// and that they have not already been specified via their position
 	for k := range e.namedArgs {
-		err = consumeKwarg(argsOptExp, e.namedArgs, k, seenKwargs)
+		err = consumeKwarg(e.namedArgs, k, argsExp[cutoff:], seenKwargs)
 		if err != nil {
 			return nil, err
 		}
 
 	}
-	err = fn.Init(e.args, e.namedArgs)
-	if err != nil {
-		return nil, err
-	}
+
+	// functions now know all non-series input args -> they know the timerange they need
```

Comment:
this comment adds confusion not clarity. I have no idea what it is trying to say.